File: Symmetric Tree - GFG/Symmetric Tree.cpp
class Solution{
    public:
    // return true/false denoting whether the tree is Symmetric or not
    bool isSymmetric(struct Node* root)
    {
	    if (root == nullptr)
            return true;

        stack<Node*> st;
        st.push(root->left);
        st.push(root->right);

        while (!st.empty()) {
            Node* t1 = st.top(); st.pop();
            Node* t2 = st.top(); st.pop();

            if (t1 == nullptr && t2 == nullptr)
                continue;
            if (t1 == nullptr || t2 == nullptr)
                return false;
            if (t1->data != t2->data)
                return false;

            st.push(t1->right); st.push(t2->left);
            st.push(t1->left); st.push(t2->right);
        }
        return true;
    }
};

File: Symmetric Tree - GFG/Symmetric Tree.java
class GfG
{
    // return true/false denoting whether the tree is Symmetric or not
    public static boolean isSymmetric(Node root)
    {
        // add your code here;
        if (root == null)
            return true;
            
        Stack<Node> st = new Stack<>();
        st.push(root.left);
        st.push(root.right);
        
        while(!st.isEmpty())
        {
            Node t1=  st.pop(), t2 = st.pop();
            if(t1==null && t2==null)  continue;
            if(t1==null || t2==null) return false;
            if(t1.data!=t2.data) return false;
            
            st.push(t1.right);    st.push(t2.left); 
             st.push(t1.left);    st.push(t2.right); 
        }
        return true;
    }
